diploma-project/components/InputFieldTimeline.jsx

```diff
@@ -0,0 +1,79 @@
+import {Collapse, InputField, Textarea} from "./index";
+
+const InputFieldTimeline = ({timeline, setTimeline}) => {
+
+    const handleAddSection = () => {
+        setTimeline([...timeline, {
+            heading: "",
+            description: "",
+            date: ""
+        }]);
+    };
+
+    const handleRemoveSection = () => {
+        setTimeline([...timeline.slice(0, timeline.length - 1)]);
+    };
+
+    return (
+        <div className="form-control space-y-5">
+            <span className="label-text">Timeline</span>
+            <ul className="space-y-3">
+                {
+                    timeline.map((section, index) => (
+                        <li key={index} className="relative">
+                            <Collapse id={1} categoryName={`Section ${index + 1}`} titleSize="text-lg" content={(
+                                <div className="pl-5">
+                                    <InputField
+                                        type="text"
+                                        labelText="Title"
+                                        placeholder=""
+                                        value={section.heading}
+                                        onChange={(e) => {
+                                            setTimeline([...timeline.slice(0, index), {
```

Comment:
This handler creates a new timeline array every time the user types a character. This could be inefficient if the timeline has a lot of sections. You might want to consider using a different state management strategy, such as using a separate state variable for each section.